File: alice/amanda/pkg/speechkit/response.go
package speechkit

type Response struct {
	Header        *Header        `json:"header,omitempty"`
	Body          *ResponseBody  `json:"response,omitempty"`
	VoiceResponse *VoiceResponse `json:"voice_response,omitempty"`
}

type ResponseBody struct {
	Cards      []Card      `json:"cards"`
	Directives []Directive `json:"directives"`
	Suggests   *Suggests   `json:"suggest"`
}

type VoiceResponse struct {
	OutputSpeech *OutputSpeech `json:"output_speech,omitempty"`
	ShouldListen *bool         `json:"should_listen,omitempty"`
}

type OutputSpeech struct {
	Type string `json:"type"`
	Text string `json:"text"`
}

type Directive struct {
	Type          string                 `json:"type"`
	Name          string                 `json:"name"`
	AnalyticsType string                 `json:"sub_name,omitempty"`
	IgnoreAnswer  *bool                  `json:"ignore_answer,omitempty"`
	Payload       map[string]interface{} `json:"payload"`
}

type Theme struct {
	ImageURL string `json:"image_url"`
}

type Button struct {
	Type       string      `json:"type"`
	Title      string      `json:"title"`
	Directives []Directive `json:"directives"`
	Text       string      `json:"text"`
	Theme      *Theme      `json:"theme"`
}

type Card struct {
	Type    string                 `json:"type"`
	Text    string                 `json:"text"`
	Body    map[string]interface{} `json:"body"`
	Buttons []Button               `json:"buttons"`
}

type Suggests struct {
	Items []Button `json:"items"`
}

File: alice/tests/library/vins_response/response.py
from abc import ABC, abstractmethod

from cached_property import cached_property


class Response(ABC):

    @property
    @abstractmethod
    def text(self):
        '''Текстовый ответ Алисы'''

    @abstractmethod
    def has_voice_response(self):
        '''Есть ли голосовой ответ Алисы'''

    @property
    @abstractmethod
    def voice_response(self):
        '''Объект голосового ответ Алисы'''

    @property
    @abstractmethod
    def output_speech_text(self):
        '''Голосовой ответ Алисы текстом'''

    @property
    @abstractmethod
    def scenario(self):
        '''Имя выигравшего сценария из аналитической информации'''

    @property
    @abstractmethod
    def product_scenario(self):
        '''Продуктовое имя сценария из аналитической информации'''

    @property
    @abstractmethod
    def intent(self):
        '''Интент сценария из аналитической информации'''

    @property
    @abstractmethod
    def directives(self):
        '''Клиентские директивы'''

    @cached_property
    def directive(self):
        return self.directives[0] if self.directives else None

    @property
    @abstractmethod
    def cards(self):
        '''Карточки Алисы: текстовы и дивные'''

    @cached_property
    def card(self):
        return self.cards[0] if self.cards else None

    @property
    @abstractmethod
    def raw(self):
        '''Доступ к сырому объекту ответа'''
